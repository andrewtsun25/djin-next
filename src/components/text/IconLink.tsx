import { Box, Typography } from "@mui/material";
import React from "react";

import { MuiNextLink, MuiNextLinkProps } from "./MuiNextLink";

interface IconLinkProps extends MuiNextLinkProps {
  icon: React.JSX.Element;
  text: string;
}

export const IconLink: React.FC<IconLinkProps> = ({
  icon,
  text,
  ...linkProps
}: IconLinkProps) => (
  <MuiNextLink {...linkProps}>
    <Box
      sx={{
        display: "flex",
        alignItems: "center",
        mb: 1,
        mt: 1,
      }}
    >
      {icon}
      <Typography
        component="span"
        sx={{
          ml: 2,
        }}
      >
        {text}
      </Typography>
    </Box>
  </MuiNextLink>
);
