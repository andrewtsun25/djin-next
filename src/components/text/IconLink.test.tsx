import BugReportIcon from "@mui/icons-material/BugReport";
import { render } from "@testing-library/react";
import IconLink from "./IconLink";

describe("IconLink", () => {
  it("renders", () => {
    const iconLink = render(
      <IconLink
        icon={<BugReportIcon />}
        text="Example Link"
        href="https://example.com"
      />
    );
    expect(iconLink).toMatchSnapshot();
  });
});
