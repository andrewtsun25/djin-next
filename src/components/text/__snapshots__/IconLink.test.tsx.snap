// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconLink renders 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="MuiBox-root css-101dtaq"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="BugReportIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 8h-2.81c-.45-.78-1.07-1.45-1.82-1.96L17 4.41 15.59 3l-2.17 2.17C12.96 5.06 12.49 5 12 5c-.49 0-.96.06-1.41.17L8.41 3 7 4.41l1.62 1.63C7.88 6.55 7.26 7.22 6.81 8H4v2h2.09c-.05.33-.09.66-.09 1v1H4v2h2v1c0 .34.04.67.09 1H4v2h2.81c1.04 1.79 2.97 3 5.19 3s4.15-1.21 5.19-3H20v-2h-2.09c.05-.33.09-.66.09-1v-1h2v-2h-2v-1c0-.34-.04-.67-.09-1H20zm-6 8h-4v-2h4zm0-4h-4v-2h4z"
            />
          </svg>
          <span
            class="MuiTypography-root MuiTypography-body1 css-1gvla80-MuiTypography-root"
          >
            Example Link
          </span>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
      href="https://example.com"
    >
      <div
        class="MuiBox-root css-101dtaq"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="BugReportIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 8h-2.81c-.45-.78-1.07-1.45-1.82-1.96L17 4.41 15.59 3l-2.17 2.17C12.96 5.06 12.49 5 12 5c-.49 0-.96.06-1.41.17L8.41 3 7 4.41l1.62 1.63C7.88 6.55 7.26 7.22 6.81 8H4v2h2.09c-.05.33-.09.66-.09 1v1H4v2h2v1c0 .34.04.67.09 1H4v2h2.81c1.04 1.79 2.97 3 5.19 3s4.15-1.21 5.19-3H20v-2h-2.09c.05-.33.09-.66.09-1v-1h2v-2h-2v-1c0-.34-.04-.67-.09-1H20zm-6 8h-4v-2h4zm0-4h-4v-2h4z"
          />
        </svg>
        <span
          class="MuiTypography-root MuiTypography-body1 css-1gvla80-MuiTypography-root"
        >
          Example Link
        </span>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
